lib/matplotlib/colors.py

```diff
@@ -703,12 +726,36 @@ def _repr_html_(self):
         """Generate an HTML representation of the Colormap."""
         png_bytes = self._repr_png_()
         png_base64 = base64.b64encode(png_bytes).decode('ascii')
-        return ('<strong>' + self.name + '</strong>' +
-                '<img ' +
+        def color_block(color):
+            hex_color = to_hex(color, keep_alpha=True)
+            return ('<div title="' + hex_color + '" ' +
+                    'style="display: inline-block; ' +
+                    'width: 1em; height: 1em; ' +
+                    'margin: 0; ' +
+                    'vertical-align: middle; ' +
+                    'border: 1px solid #555; ' +
+                    'background-color: ' + hex_color + ';">' +
+                    '</div>')
+
+        return ('<div style="vertical-align: middle;">' +
+                '<strong>' + self.name + '</strong> ' +
+                '</div>' +
+                '<div class="cmap"><img ' +
                 'alt="' + self.name + ' color map" ' +
-                'title="' + self.name + '"' +
+                'title="' + self.name + '" ' +
                 'style="border: 1px solid #555;" ' +
-                'src="data:image/png;base64,' + png_base64 + '">')
+                'src="data:image/png;base64,' + png_base64 + '"></div>' +
+                '<div style="vertical-align: middle; max-width: 402px; ' +
```

Comment:
Minor question:
I was thinking about this 402 value some more. The reason it is 400 (+ 1 pixel on each edge) is due to `_repr_png` sampling the colormap at that spacing. I'm wondering if it would make more sense to sample the colormap at 256 (or 512, or cmap.N) spacing (since that is usually how many values are in the colormaps) and then set the div image width to 400 instead? That would allow each cmap value to be sampled the same number of times and then the monitor/printer can decide how to resample the image pixels for display onto different widths?

Reply:
@greglucas @timhoffm Good suggestion. I changed it from 400 pixels to 512 pixels since 256 is the most common number of colors and that divides evenly. I think it's important to keep a fixed width (and not use `cmap.N`) in case users wish to save the images and use them outside the browser (I know I have use cases for that, at least, and that's part of what motivated me to make this feature). I also made the size into a private module constant `_REPR_PNG_SIZE` that can be used in both `_repr_png_` and `_repr_html_`. I tested in JupyterLab and the HTML representation scales down fluidly in browser windows with small widths.